src/main/java/hudson/plugins/s3/Entry.java

```diff
@@ -49,6 +49,11 @@
      */
     public boolean noUploadOnFailure;
 
+    /**
+     * Do not publish the artifacts when build is aborted
+     */
+    public boolean noUploadOnAborted;
```

Comment:
To the best of my understanding, this needs to be `true` in order to preserve the backwards compatibility. However, existing configurations will be migrated so this gets set to `false` (default value for `boolean`) causing it to change semantics.

Better use `Boolean` here and do the `null -> true` conversion in `readResolve`: https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility#Hintonretainingbackwardcompatibility-Scenario:Addinganewfield

Reply:
Let me know what you think